chainntnfs/txconfnotifier.go

```diff
@@ -177,91 +188,106 @@ func (tcn *TxConfNotifier) UpdateConfDetails(txid chainhash.Hash,
 
 	// First, we'll determine whether we have an active notification for
 	// this transaction with the given ID.
-	ntfns, ok := tcn.confNotifications[txid]
+	confSet, ok := tcn.confNotifications[txid]
 	if !ok {
-		return fmt.Errorf("no notifications found for txid %v", txid)
+		return fmt.Errorf("no notification found with TxID %v", txid)
 	}
 
-	ntfn, ok := ntfns[clientID]
+	ntfn, ok := confSet.ntfns[clientID]
 	if !ok {
 		return fmt.Errorf("no notification found with ID %v", clientID)
 	}
 
+	// Decrement the rescan count for each registered client that finishes.
+	confSet.rescanCount--
+
 	// If the notification has already recognized that the transaction
 	// confirmed, there's nothing left for us to do.
 	if ntfn.details != nil {
 		return nil
 	}
 
-	err := tcn.hintCache.CommitConfirmHint(details.BlockHeight, txid)
-	if err != nil {
-		// The error is not fatal, so we should not return an error to
-		// the caller.
-		Log.Errorf("Unable to update confirm hint to %d for %v: %v",
-			details.BlockHeight, txid, err)
-	}
-
 	// The notifier has yet to reach the height at which the transaction was
 	// included in a block, so we should defer until handling it then within
 	// ConnectTip.
 	if details == nil || details.BlockHeight > tcn.currentHeight {
 		return nil
 	}
 
-	ntfn.details = details
-
-	// Now, we'll examine whether the transaction of this notification
-	// request has reched its required number of confirmations. If it has,
-	// we'll disaptch a confirmation notification to the caller.
-	confHeight := details.BlockHeight + ntfn.NumConfirmations - 1
-	if confHeight <= tcn.currentHeight {
-		Log.Infof("Dispatching %v conf notification for %v",
-			ntfn.NumConfirmations, ntfn.TxID)
+	err := tcn.hintCache.CommitConfirmHint(details.BlockHeight, txid)
+	if err != nil {
+		// The error is not fatal, so we should not return an error to
+		// the caller.
+		Log.Errorf("Unable to update confirm hint to %d for %v: %v",
+			details.BlockHeight, txid, err)
+	}
 
-		// We'll send a 0 value to the Updates channel, indicating that
-		// the transaction has already been confirmed.
-		select {
-		case ntfn.Event.Updates <- 0:
-		case <-tcn.quit:
-			return ErrTxConfNotifierExiting
+	// Update the conf details of all ntfns that don't yet have them.
+	for _, ntfn := range confSet.ntfns {
```

Comment:
Isn't this already handled by each rescan calling `UpdateConfDetails` for its client ID?

Reply:
only one rescan is dispatched now